cmd/traffic/cmd/manager/service.go

```diff
@@ -545,3 +546,33 @@ func (m *Manager) WatchLookupHost(session *rpc.SessionInfo, stream rpc.Manager_W
 func (m *Manager) expire() {
 	m.state.ExpireSessions(m.clock.Now().Add(-15 * time.Second))
 }
+
+// reapDomain informs SystemA that an intercept with a domain has been garbage collected
+func (m *Manager) reapDomain(ctx context.Context, sa systema.SystemACRUDClient, interceptUpdate watchable.InterceptMapUpdate) error {
+	dlog.Debugf(ctx, "systema: removing domain: %q", interceptUpdate.Value.PreviewDomain)
+	_, err := sa.RemoveDomain(ctx, &systema.RemoveDomainRequest{
+		Domain: interceptUpdate.Value.PreviewDomain,
+	})
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
we could probably just `return err` here; same for `reapIntercept`

Reply:
Yeah I went back and forth on that, my concern was that it'd potentially be confusing to read cuz at a quick glance you might think it's always returning an `err`, and felt it was more clear to only return `err` when `err != nil`.

Reply:
I tend to prefer the explicit form like Donny originally did.  I have two reasons for this preference:

- It makes coverage testing easier, as you can see in the coverage report whether you covered both the error and non-error cases.  The single `return err` hides this info.
- Because nil structs get silently promoted to non-nil interfaces, having the separate returns is defensive programming in that it catches a class of bugs.  For example:

Consider the code

```go
func innerFunc() error {
	…
}

func outerFunc() error {
	err := innerFunc()
	return err
}
```

In this version, that `return err` is fine.  However, what if we changed `innerFunc`'s signature changed to `func innerFunc() *MyErrorType` (where `*MyErrorType` implements `error`)?  It seems like an innocuous change, and the compiler would happily accept it without complaining.  However, the nil `*MyErrorType` would get promoted to a non-nil `error`, and suddenly `outerFunc` is returning non-nil when it should be returning nil!

And things like that give me the willies, so I prefer to write code in a style where that can't happen.

Reply:
> Because nil structs get silently promoted to non-nil interfaces, having the separate returns is defensive programming in that it catches a class of bugs.

Waat I had no idea that could happen, makes sense to keep it the way Donny had it in that case. The point about code coverage is valid too